framework/tlvf/test/tlvf_test.cpp

```diff
@@ -276,34 +359,34 @@ int test_all()
     if (!thirdTlv->set_manufacturer("Intel"))
         return false;
 
-    // All attributes have been initialized to default.
-    LOG(DEBUG) << "Create WSC M2 encrypted settings";
-    auto encrypted_settings_tx = thirdTlv->create_encrypted_settings();
-    if (!encrypted_settings_tx) {
-        LOG(ERROR) << "Failed to create encrypted settings";
-        return false;
-    }
-
-    encrypted_settings_tx->set_ssid("test_ssid");
-    encrypted_settings_tx->authentication_type_attr().data = WSC::WSC_AUTH_WPA2; //DUMMY
-    encrypted_settings_tx->encryption_type_attr().data     = WSC::WSC_ENCR_AES;
-    std::fill(encrypted_settings_tx->network_key_attr().data,
-              encrypted_settings_tx->network_key_attr().data +
-                  encrypted_settings_tx->network_key_attr().data_length,
-              0xaa); //DUMMY
-    std::fill(encrypted_settings_tx->key_wrap_auth_attr().data,
-              encrypted_settings_tx->key_wrap_auth_attr().data +
-                  encrypted_settings_tx->key_wrap_auth_attr().data_length,
-              0xff); //DUMMY
-    LOG(DEBUG) << "WSC encrypted_settings_tx:" << std::endl
-               << "     ssid: " << encrypted_settings_tx->ssid() << std::endl
-               << "     authentication_type: "
-               << encrypted_settings_tx->authentication_type_attr().data << std::endl
-               << "     encryption_type: " << encrypted_settings_tx->encryption_type_attr().data
-               << std::endl;
-    LOG(DEBUG) << "Add WSC M2 encrypted settings";
-    if (!thirdTlv->add_encrypted_settings(encrypted_settings_tx)) {
-        LOG(ERROR) << "Failed to add encrypted settings";
+    /**
+     * @brief Diffie helman key exchange
+     * 
+     */
+    uint8_t key1[192];
+    uint8_t key2[192];
+    unsigned key1_length = sizeof(key1);
+    unsigned key2_length = sizeof(key2);
+    std::fill(key1, key1 + key1_length, 1);
+    std::fill(key2, key2 + key2_length, 2);
+    mapf::encryption::diffie_hellman m1;
+    mapf::encryption::diffie_hellman m2;
+    // diffie-helman key-exchange
+    m1.compute_key(key1, key1_length, m2.pubkey(), m2.pubkey_length());
+    m2.compute_key(key2, key2_length, m1.pubkey(), m1.pubkey_length());
+    key1_length = sizeof(key1);
+    uint8_t m1_nonce[192];
+    uint8_t m2_nonce[192];
+    mapf::encryption::create_nonce(m1_nonce, sizeof(m1_nonce));
+    mapf::encryption::create_nonce(m2_nonce, sizeof(m2_nonce));
+    uint8_t mac[6] = {0x11, 0x22, 0x33, 0x44, 0x55, 0x66};
+    uint8_t authkey[32];
+    uint8_t keywrapkey[16];
+    uint8_t iv[16];
```

Comment:
Again, there's a lot of duplication here from the actual implementation.

I think we really would want an extra class that does all this functionality, i.e. move it out of the master_thread and slave_thread. This would also solve part of the lifetime issue of the diffie_hellman class, and it could also store the nonce.

It would necessarily need to have access to the M1 and M2 definitions, so encryption.h/cpp is not the appropriate place. But then I'm not sure what is the appropriate place...

Note that this can be done in a future refactoring PR. In other words: (nitpick).

Reply:
We can move it to the TLVF as none generated files, like BaseClass.cpp.
Why not add autoconfig class in the framework as part of mapfcommon.so? It means that mapfcommon.so will need to link to tlvf.so, is that an issue?

Reply:
Anyway I will do that in a separate commit, after we finish the first review iteration.

Reply:
> We can move it to the TLVF as none generated files, like BaseClass.cpp.
> Why not add autoconfig class in the framework as part of mapfcommon.so? It means that mapfcommon.so will need to link to tlvf.so, is that an issue?

I thought tlvf.so already depended on mapfcommon, but apparently it doesn't so indeed we can put it in mapfcommon.